src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.Strategies
+{
+    internal static partial class FileStreamHelpers
+    {
+        // It's enabled by default. We are going to change that once we fix #16354, #25905 and #24847.
+        internal static bool UseLegacyStrategy { get; } = GetLegacyFileStreamSetting();
+
+        private static bool GetLegacyFileStreamSetting()
+        {
+            if (AppContext.TryGetSwitch("System.IO.UseNet5CompatFileStream", out bool fileConfig))
+            {
+                return fileConfig;
+            }
+
+            string? envVar = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_IO_USENET5COMPATFILESTREAM");
+            return envVar is null
+                ? true // legacy is currently enabled by default;
+                : bool.IsTrueStringIgnoreCase(envVar) || envVar.Equals("1");
+        }
+
+        internal static FileStreamStrategy ChooseStrategy(FileStream fileStream, SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync)
+            => WrapIfDerivedType(fileStream, ChooseStrategy(handle, access, bufferSize, isAsync));
```

Comment:
Nit: can we please use other name for the OS-specific method? e.g:
```suggestion
            => WrapIfDerivedType(fileStream, ChooseStrategyCore(handle, access, bufferSize, isAsync));
```